### ISSUE TITLE
How to load init script before Stryker run

### ISSUE THREAD

**Issue by Saro-DA** [stryker-mutator/stryker#1367]
Hi. I have one init script which is required to load some env variables before executing mocha. Here's that script in my package.json.

...
"test": "nyc mocha -r ./tests/setup-mocha.js --recursive tests/unittests/**/*.test.js --exit",
"test-stryker": "mocha -r ./tests/setup-mocha.js tests/unittests/schemas/someSchema.test.js --exit",
...

Now that I want to run Stryker, this setup-mocha.js is not loaded anymore and my env variables are undefined.

module.exports = function(config) {
  config.set({
    mutator: "javascript",
    packageManager: "npm",
    reporters: ["clear-text", "progress"],
    testRunner: "command",
	commandRunner: {
		command: "npm run test-stryker",
	},
    transpilers: [],
    mutate: ["schemas/someSchema.js"]
  });
};

How to force that script to be loaded before starting the Stryker?
Thanks in advance.

**Comment by nicojs** [comment]
Hi @Saro-DA thanks for opening an issue.

Your config looks fine to me. Could you give some more info?

* Does `npm run test-stryker` work as expected?
* Does it also work if you run it from inside the sandbox directory?  
  The sandbox directory can be found in the `.stryker-tmp` folder. You should be able to just run `npm run test-stryker` from there as well

**Comment by Saro-DA** [comment]
Hi @nicojs. Thanks for replying. 

Yes it does work as expected. And even the `Stryker run` works fine if I manually set those environment variables inside my `*.test.js` files.
Inside my `setup-mocha.js` file, I'm using `dotenv` module to load environment variables from a config file. And the issue seems to be that this module is not able to do that when is executed with 'Stryker run'. It works fine with `npm run test-stryker` though.

**Comment by nicojs** [comment]
> I'm using `dotenv` module to load environment variables from a config file.

Aha, I think that is your smoking gun. Is your `.env` file included in the list of files inside sandboxes?

To see that, run `npx stryker run --logLevel debug` and look for: `18:19:11 (8560) DEBUG InputFileResolver All input files: [`

There, all files are listed. Is your `.env` one of them?

**Comment by simondel** [comment]
#1343 might be a duplicate of this issue. It contains a workaround.

**Comment by Mrna1** [comment]
@nicojs, @simondel: Is there any way how to execute command before standard stryker run. Basically, I want to convert paths form @account to ../../account. I can do it by script, but it will modify the files. I would like to do it in a sandbox before the mutation starts. 
commandRunner: {
            command: "npm run modify-paths",
        },
it doesn't work because it is just an alternative for testRunner.

Thank you

**Comment by simondel** [comment]
How do you run your tests right now @Mrna1? You may be able to use that command in the test command test runner:
```
commandRunner: {
command: "npm run modify-paths | npm test",
},
```
(the command above will probably only work on windows. Perhaps `npm test` is good enough for your use case)

**Comment by Mrna1** [comment]
@simondel: Thank you, unfortunately, it fails, because the paths are not modified :( So there possibility to get logs from my modification functions? Maybe there are different paths in the sandbox

11:33:17 (25716) DEBUG ChildProcessProxy Disposing of worker process 7916
11:33:17 (25716) DEBUG ChildProcessProxy Kill 7916

**Comment by simondel** [comment]
I don't know what your modifications do exactly and what they usually log. You could try running stryker with log level 'trace': https://www.npmjs.com/package/@stryker-mutator/core#loglevel--string-
